core/templates/dev/head/pages/exploration_editor/editor_tab/StateSolution.js

```diff
@@ -0,0 +1,277 @@
+    SolutionObjectFactory, responsesService, AnswerClassificationService,
+    explorationContextService, angularNameService) {
+    $scope.editorContextService = editorContextService;
+    $scope.stateSolutionService = stateSolutionService;
+    $scope.isActive = false;
+    $scope.answerCheckPassed = false;
+    $scope.INTERACTION_SPECS = INTERACTION_SPECS;
+    $scope.interactionHtml = '';
+
+    $scope.stateHintsService = stateHintsService;
+    $scope.stateSolutionService = stateSolutionService;
+    $scope.correctAnswer = null;
+
+    $scope.submitAnswer = function (answer) {
+      $scope.correctAnswer = answer;
+    };
+
+    $scope.toggleIsActive = function() {
+      $scope.isActive = !$scope.isActive;
+    };
+
+    $scope.$on('stateEditorInitialized', function(evt, stateData) {
+      stateSolutionService.init(
+        editorContextService.getActiveStateName(),
+        stateData.interaction.solution);
+      $scope.interactionHtml = (
+        oppiaExplorationHtmlFormatterService.getInteractionHtml(
+          stateInteractionIdService.savedMemento,
+          explorationStatesService.getInteractionCustomizationArgsMemento(
+            editorContextService.getActiveStateName()),
+          'currentInteractionHtml'));
+
+      $scope.solutionIsSpecified = (
+        !!JSON.stringify(stateSolutionService.savedMemento));
+      $scope.isActive = false;
+      var interactionId = stateInteractionIdService.savedMemento;
+
+      $scope.INTERACTION_SPECS = INTERACTION_SPECS[interactionId];
+
+      if (StateSolutionHelperService.isSupportedInteraction(interactionId)) {
+        $scope.interactionHtml = null;
+        $scope.objectType = (
+          StateSolutionHelperService.getUnsupportedInteractionObjectType(
+            interactionId));
+      } else {
+        $scope.objectType = null;
+      }
+    });
+
+    $scope.getSolutionSummary = function() {
+      var solution = stateSolutionService.displayed;
+      var isExclusiveAnswer = (
+        solution.answerIsExclusive ? 'Only' : 'One');
+      var correctAnswer = StateSolutionHelperService.getCorrectAnswer(solution);
+      var explanation = (
+        $filter('convertToPlainText')(solution.explanation));
+      return (
+        '[' + isExclusiveAnswer + ' solution is ' + correctAnswer + '] ' +
+        explanation);
+    };
+
+    // This returns false if the current interaction ID is null.
+    $scope.isCurrentInteractionLinear = function() {
+      var interactionId = stateInteractionIdService.savedMemento;
+      return interactionId && INTERACTION_SPECS[interactionId].is_linear;
+    };
+
+    $scope.openAddSolutionModal = function() {
+      alertsService.clearWarnings();
+      $rootScope.$broadcast('externalSave');
+      $scope.isActive = false;
+
+      $modal.open({
+        templateUrl: 'modals/addSolution',
+        backdrop: 'static',
+        scope: $scope,
+        controller: [
+          '$scope', '$rootScope', '$modalInstance', 'editorContextService',
```

Comment:
$rootScope should not be used here.

Do you even use editorContextService in this modal, btw? I see it outside, but not inside...